### PR TITLE
Feature/gdev83

### PR DESCRIPTION
**Pull request by KerstenBreuer**
Add examples for unit and integrations tests.

Please note, (as #3) this PR depends on the shared lib at https://github.com/ghga-de/ghga-service-chassis-lib/tree/feature/rest which is not yet available at PyPI. Thus to test it now run the following:

```
pip install git+https://github.com/ghga-de/ghga-service-chassis-lib@feature/rest
dev_install
```

This is also why the CI currently fails. Once the shared library is realized, the installation will work as intended.

This PR depends on #1 and #5, please review them first.

### CODE REVIEW

File: tests/unit/fixtures/__init__.py [hunk 1]
```diff
@@ -13,4 +13,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Fixtures that exclusively used in integration tests"""
+"""Fixtures that exclusively used in unit tests"""
+
+import os
+
+BASE_DIR = os.path.dirname(__file__)
```
Comment (ac-jorellanaf) [inline tests/unit/fixtures/__init__.py:20]: Pathlib?

File: tests/unit/test_core_greeting.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+from my_microservice.models import Greeting
+from .fixtures.greetings import greeting_expression_fixture
+
+
+@pytest.mark.parametrize(
+    "language,isinformal,exception_expected",
+    [
+        ("Greek", True, None),
+        ("Greek", False, None),
+        ("Croatian", True, ValueError),
```
Comment (ac-jorellanaf) [inline tests/unit/test_core_greeting.py:31]: So Croatian does not have informal greetings, so if you get one that is, you get a Value Error?

**Review by ac-jorellanaf — APPROVED** [review]

File: tests/unit/fixtures/__init__.py (hunk 1 above)
Comment (KerstenBreuer) [reply tests/unit/fixtures/__init__.py:20]: Oh yes sure, please go ahead and change it

File: tests/unit/test_core_greeting.py (hunk 2 above)
Comment (KerstenBreuer) [reply tests/unit/test_core_greeting.py:31]: Yes exactly, so of course Croatian has informal greetings, but we haven't configured them this is why we expect a value error. Again just a stupid example for this toy service.

**Review by lkuchenb — APPROVED** [review]
Code lgtm.

I don't think I quite understand the use of fixtures as replacements for the values defined in the main code though. If they're part of the code it would be fine to rely on them in the tests and also that changes to the code would break the tests. Or put differently, like that a part of the code (i.e. the hard coded values) are not begin tested.

**Comment by KerstenBreuer** [comment]
> Code lgtm.
> 
> I don't think I quite understand the use of fixtures as replacements for the values defined in the main code though. If they're part of the code it would be fine to rely on them in the tests and also that changes to the code would break the tests. Or put differently, like that a part of the code (i.e. the hard coded values) are not begin tested.

Hi @lkuchenb, thanks for your feedback. Do you refer to the GreetingExpressions? I think I understand what you mean. Because they are part of the code it might be enough to just check whether the functions are working with the hard-coded option. That would be a good integration test. But for a unit test, you might be more interested in testing individual functions and see whether they can handle all potential edge cases independent of the specific context in which they are used now. So you would like to test, that they will even work if you change the rest of the code including the hard-coded greetings. It's about thinking of each function as an independent component, almost like a microservice. You know what I mean?
